File: backend/src/main/java/com/onair/hearit/admin/dto/response/RecommendHearitResponse.java
package com.onair.hearit.admin.dto.response;

import com.onair.hearit.domain.Hearit;
import java.time.LocalDate;

public record RecommendHearitResponse(
        Long id,
        String title,
        String summary,
        Integer playTime,
        String categoryName,
        LocalDate createdAt,
        LocalDate lastRecommendedDate
) {

    public static RecommendHearitResponse of(Hearit hearit, LocalDate lastRecommendedDate) {
        return new RecommendHearitResponse(
                hearit.getId(),
                hearit.getTitle(),
                hearit.getSummary(),
                hearit.getPlayTime(),
                hearit.getCategory().getName(),
                LocalDate.from(hearit.getCreatedAt()),
                lastRecommendedDate
        );
    }
}

File: android/app/src/main/java/com/onair/hearit/data/RecommendHearitResponse.kt
package com.onair.hearit.data

import com.onair.hearit.domain.RecommendHearitItem
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class RecommendHearitResponse(
    @SerialName("createdAt")
    val createdAt: String,
    @SerialName("id")
    val id: Long,
    @SerialName("playTime")
    val playTime: Int,
    @SerialName("source")
    val source: String,
    @SerialName("summary")
    val summary: String,
    @SerialName("title")
    val title: String,
)

fun RecommendHearitResponse.toDomain(): RecommendHearitItem =
    RecommendHearitItem(
        id = this.id,
        title = this.title,
        desc = this.summary,
    )
